src/app/(home)/Table/Filters.tsx

```diff
@@ -3,43 +3,136 @@ import {
   DropdownMenu,
   DropdownMenuCheckboxItem,
   DropdownMenuContent,
+  DropdownMenuLabel,
+  DropdownMenuRadioGroup,
+  DropdownMenuRadioItem,
+  DropdownMenuSeparator,
   DropdownMenuTrigger,
 } from '@/components/ui/dropdown-menu';
-import { Input } from '@/components/ui/input';
-import { Session } from '@/types';
+import { ApiFormOptions, FilterSchema, Session } from '@/types';
 import { type Table } from '@tanstack/react-table';
-import { Settings2, X } from 'lucide-react';
+import { Filter, Settings2, X } from 'lucide-react';
 import Link from 'next/link';
 import { useMemo } from 'react';
 
-const Filters = ({ table }: { table: Table<Session> }) => {
-  const isFiltered = useMemo(() => table.getState().columnFilters.length > 0, [table]);
+const Filters = ({ table, apiOptions }: { table: Table<Session>; apiOptions: ApiFormOptions }) => {
+  const isFiltered = useMemo(
+    () => table.getState().columnFilters.length > 0,
+    [table.getState().columnFilters.length]
+  );
+
+  const filterColumns = useMemo(() => {
+    const fields: FilterSchema = {
+      group: {
+        name: 'group',
+        label: 'Group',
+        options: apiOptions.group ?? [],
+        show: Boolean(apiOptions.group?.length),
+        onChange: (value: string) => {
+          table.getColumn('group')?.setFilterValue(value);
+          table.getColumn('parentId')?.setFilterValue('');
+          table.getColumn('batchId')?.setFilterValue('');
+          table.resetPageIndex(true);
+        },
+      },
+      parentId: {
+        name: 'parentId',
+        label: 'Parent Batch',
+        options: [],
+        show: false,
+        onChange: (value: string) => {
+          table.getColumn('parentId')?.setFilterValue(value);
+          table.getColumn('batchId')?.setFilterValue('');
+          table.resetPageIndex(true);
+        },
+      },
+      batchId: {
+        name: 'batchId',
+        label: 'Quiz Batch',
+        options: [],
+        show: false,
+        onChange: (value: string) => {
+          table.getColumn('batchId')?.setFilterValue(value);
+          table.resetPageIndex(true);
+        },
+      },
+    };
+    const groupFilter = table.getColumn('group')?.getFilterValue();
+    const parentIdFilter = table.getColumn('parentId')?.getFilterValue();
+
+    if (groupFilter) {
+      const authGroupId = apiOptions.group?.find((item) => item.value === groupFilter)?.id;
+      const filteredQuizBatchOptions =
+        apiOptions?.batch?.filter((item) => item.groupId === authGroupId && !item.parentId) ?? [];
+      fields.parentId.options = filteredQuizBatchOptions;
+      fields.parentId.show = true;
+
+      if (parentIdFilter) {
+        const quizBatchId = apiOptions.batch?.find((item) => item.value === parentIdFilter)?.id;
+        const filteredClassBatchOptions =
+          apiOptions?.batch?.filter((item) => item.parentId === quizBatchId) ?? [];
```

Comment:
how come classBatchId not being used

Reply:
Here I'm setting quiz batch options for filters using authGroup. 
and class batch options using the selected quizBatch

Their is no filter which depends on the classBatchId, that's why its not being used.